hack/test_e2e.go

```diff
@@ -0,0 +1,944 @@
+
+	if ! command_exist("bpf2go") {
+		if ! command_exist("go") {
+			fmt.Println("Dependency not met: 'bpf2go' not installed and cannot install with 'go'")
+			os.Exit(1)
+		} 
+		
+		fmt.Println("'bpf2go' not found, installing with 'go'")
+		_, err := os.Stat(install_directory)
+		if err != nil && os.IsNotExist(err) {
+			log.Fatal(err)
+		}
+		// set GOBIN to bin_directory to ensure that binary is in search path	
+		_ = os.Setenv("GOBIN", install_directory)
+		// remove GOPATH just to be sure
+		_ = os.Setenv("GOPATH", "")
+
+		cmd := exec.Command("go", "install", "github.com/cilium/ebpf/cmd/bpf2go@v0.9.2")
+		err = cmd.Run()
+		if_error_exit(err)
+	
+		var buffer bytes.Buffer
+		cmd = exec.Command("which", "bpf2go")
+		cmd.Stdout = &buffer
+		err = cmd.Run()
+		if_error_exit(err)
+		fmt.Printf("The tool bpf2go is installed at: %s\n", buffer.String())	
+	} 
+} 
+
+
+
+func install_binaries(bin_directory, k8s_version, operating_system, base_dir_path string, ) {
+    
+    // Description:
+    // Copy binaries from the net to binaries directory
+    //
+    // Arguments:
+    //   arg1: binary directory, path to where ginko will be installed
+    //   arg2: Kubernetes version
+    //  arg3: OS, name of the operating system
+	/////////////////////////////////////////////////////////////////////////////////
+	wd, err := os.Getwd()
+	if_error_exit(err)
+	
+	if wd != base_dir_path {
+		err = os.Chdir(base_dir_path)
+		if_error_exit(err)
+	}
+	err = os.MkdirAll(bin_directory, 0755) 
+
+	add_to_path(bin_directory) 
+
+	setup_kind(bin_directory, operating_system)
+	setup_kubectl(bin_directory, k8s_version, operating_system)
+	setup_ginkgo(bin_directory, k8s_version, operating_system)
+	setup_bpf2go(bin_directory)
+}
+
+func detect_container_engine() {
+	///////////////////////////////////////////////////////////////////////////
+    // Description:                                                           
+    // Detect Container Engine, by default it is docker but developers might   
+    // use real alternatives like podman. The project welcome both.            
+    //                                                                         
+    // Arguments:                                                              
+    //   None
+	///////////////////////////////////////////////////////////////////////////                                                                  	
+	
+	CONTAINER_ENGINE = "docker"
+	cmd := exec.Command(CONTAINER_ENGINE) 
+	err := cmd.Run()
+	if err != nil {
+		// If docker is not available, let's check if podman exists
+		CONTAINER_ENGINE = "podman"
+		cmd = exec.Command(CONTAINER_ENGINE, "--help")
+		err = cmd.Run() 
+		if err != nil {
+			fmt.Println("The e2e tests currently support docker and podman as the container engine. Please install either of them")
+			os.Exit(1)
+		}
+	}
+	fmt.Println("Detected Container Engine:", CONTAINER_ENGINE)
+}
+
+func container_build(CONTAINER_FILE string, ci_mode bool) {
```

Comment:
best to use `containerFile` and `ciMode` as parameters, following go conventions

https://go.dev/doc/effective_go#:~:text=Finally%2C%20the%20convention%20in%20Go%20is%20to%20use%20MixedCaps%20or%20mixedCaps%20rather%20than%20underscores%20to%20write%20multiword%20names.

Reply:
Done